internal/cmd/egctl/translate.go

```diff
@@ -792,6 +793,10 @@ func kubernetesYAMLToResources(str string, addMissingResources bool) (*gatewayap
 				},
 				Spec: typedSpec.(corev1.ServiceSpec),
 			}
+			// fill with dummy IP when service clusterIP is empty
+			if len(service.Spec.ClusterIP) == 0 {
```

Comment:
imo we should only do this if `--add-missing-resources` is set

Reply:
sounds right, added